### PR TITLE
Semver-major developments

### PR DESCRIPTION
**Pull request by RyanZim**
This is a PR of the long-running `develop` branch against `master`. I've rebased the branch onto master, so there should be no merge trouble. @mikelambert Please merge and release either v0.16.0 or v1.0.0.

CHANGELOG for release:
```
- **BREAKING:** Drop support for Node 4
- Upgrade to jsdom v11.3.0
- Fix a bug in handling the `animation` property ([#284](https://github.com/giakki/uncss/issues/284))
- Remove bluebird dependency
```

### CODE REVIEW

File: src/jsdom.js [hunk 1]
```diff
@@ -1,22 +1,9 @@
...
-// Configure.
-const jsdomAsync = Promise.promisify(jsdom.env, { context: jsdom });
-
-/**
- * Closes a page.
- * @param {Object} Page opened by jsdom
- * @return {void}
- */
-function cleanup(page) {
-    return page.close();
```
Comment (mikelambert) [inline src/jsdom.js]: Where does this line go, is anything cleaning up our jsdom pages in the new codebase? Or if not, why is it no longer needed?

**Comment by RyanZim** [comment]
Need to fix merge conflicts here, will do tomorrow hopefully; @mikelambert feel free to manually merge and fix if you prefer.

**Comment by mikelambert** [comment]
Went through the diffs, great thorough work on the conversion!

I'd merge but there's some merge conflict on `package-lock.json` I'm not quite sure the right way to resolve.

**Comment by RyanZim** [comment]
@mikelambert Rebased and fixed conflicts. Please merge and release.

**Comment by mikelambert** [comment]
Sorry for the delay, but released as 0.16.0.

**Pull request merged**